### PR TITLE
test(performance): status-access-vms

### PR DESCRIPTION
**Pull request by universal-itengineer**
## Description

This is part of the performance tests.
## Checklist
- [ ] The code is covered by unit tests.
- [ ] e2e tests passed.
- [ ] Documentation updated according to the changes.
- [ ] Changes were tested in the Kubernetes cluster manually.

### CODE REVIEW

File: tests/performance/status-access-vms/ansible/run.sh [hunk 1]
```diff
@@ -0,0 +1,48 @@
+#!/bin/bash
+SSKKEY="../sshkeys/id_ed"
+NAMESPACE=test-perf
+
+function prepaer_ssh_key {
```
Comment (fl64) [inline tests/performance/status-access-vms/ansible/run.sh:5]:
```suggestion
function prepare_ssh_key {
```

File: tests/performance/status-access-vms/ansible/run.sh [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+all:
+  hosts:" > $inventory_file
+
+    while IFS=$'\t' read -r vm_name vm_ip; do
+        echo "    ${vm_name}.${NAMESPACE}:" >> $inventory_file
+    done <<< "$vms"
+}
+
+function main {
+    prepaer_ssh_key
```
Comment (fl64) [inline tests/performance/status-access-vms/ansible/run.sh:31]:
```suggestion
    prepare_ssh_key
```

File: tests/performance/status-access-vms/ansible/ansible.cfg [hunk 3]
```diff
@@ -0,0 +1,28 @@
+[defaults]
+ansible_managed=Ansible managed: {file} modified on %Y-%m-%d %H:%M:%S
+inventory=$PWD/inventory
+library=$PWD/modules:$HOME/ansible/modules
+collections_path=$PWD/collections
```
Comment (fl64) [inline tests/performance/status-access-vms/ansible/ansible.cfg:5]: Do we need this?

File: tests/performance/status-access-vms/ansible/ansible.cfg [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+transport=smart
+host_key_checking=False
+fact_caching=jsonfile
+fact_caching_connection=$PWD/tmp
+fact_caching_timeout=3600
+
+gathering=smart
+
+bin_ansible_callbacks=True
+deprecation_warnings=Talse
```
Comment (fl64) [inline tests/performance/status-access-vms/ansible/ansible.cfg:19]:
```suggestion
deprecation_warnings=false
```

File: tests/performance/status-access-vms/ansible/ansible.cfg [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+forks=500
+transport=smart
+host_key_checking=False
+fact_caching=jsonfile
+fact_caching_connection=$PWD/tmp
+fact_caching_timeout=3600
+
+gathering=smart
+
+bin_ansible_callbacks=True
```
Comment (fl64) [inline tests/performance/status-access-vms/ansible/ansible.cfg:18]:
```suggestion
bin_ansible_callbacks=true
```

File: tests/performance/status-access-vms/ansible/ansible.cfg [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+ansible_managed=Ansible managed: {file} modified on %Y-%m-%d %H:%M:%S
+inventory=$PWD/inventory
+library=$PWD/modules:$HOME/ansible/modules
+collections_path=$PWD/collections
+roles_path=$PWD/roles
+interpreter_python=python3
+
+forks=500
+transport=smart
+host_key_checking=False
```
Comment (fl64) [inline tests/performance/status-access-vms/ansible/ansible.cfg:11]: Please replace all booleans with lowercase letters

File: tests/performance/status-access-vms/ansible/run.sh [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+function generate_inventory {
+    vms=$(kubectl -n $NAMESPACE get vm -o=jsonpath='{range .items[*]}{.metadata.name}{"\t"}{.status.ipAddress}{"\n"}{end}')
+    mkdir -p inventory
+    inventory_file="inventory/hosts.yml"
+    echo "---
+all:
+  hosts:" > $inventory_file
+
+    while IFS=$'\t' read -r vm_name vm_ip; do
+        echo "    ${vm_name}.${NAMESPACE}:" >> $inventory_file
```
Comment (fl64) [inline tests/performance/status-access-vms/ansible/run.sh:26]: why not VM_NAME?

File: tests/performance/status-access-vms/sshkeys/id_ed [hunk 8]
```diff
@@ -0,0 +1,7 @@
+-----BEGIN OPENSSH PRIVATE KEY-----
```
Comment (fl64) [inline tests/performance/status-access-vms/sshkeys/id_ed:1]: Do we really need to keep it here?

**Comment by fl64** [comment]
I have a suggestion to add a taskfile with the following tasks:
- run-ansible
- run-tank

**Review by fl64 — CHANGES_REQUESTED** [review]

File: tests/performance/status-access-vms/ansible/run.sh [hunk 9]
```diff
@@ -0,0 +1,152 @@
...
+        HOSTS_UNREACHABLE=$(( $(grep -E 'unreachable=[1-9]+' $ANSIBLE_REPORT_FILE | wc -l) ))
+        OK_PCT=$(bc -l <<< "scale=2; $HOSTS_OK/$HOSTS_TOTAL*100")
+        
+        if [[ $HOSTS_UNREACHABLE -ne 0 ]]; then
+            grep -E 'unreachable=[1-9]+' $ANSIBLE_REPORT_FILE
+        fi
+
+        echo "OK hosts count:$HOSTS_OK pct.:$OK_PCT% | Unreachable hosts $HOSTS_UNREACHABLE | Total hosts $HOSTS_TOTAL"
+        echo "Wait 2 sec"
+        echo -e "---\n"
```
Comment (diafour) [inline tests/performance/status-access-vms/ansible/run.sh:147]:
```suggestion
        echo "---"
```

**Review by diafour — APPROVED** [review]

**Review by fl64 — APPROVED** [review]

**Pull request merged**